jujugui/static/gui/src/app/components/entity-details/entity-details.js

```diff
@@ -41,22 +42,30 @@ YUI.add('entity-details', function() {
       @return {Object} A generated state object which can be passed to setState.
     */
     generateState: function(nextProps) {
-      var state = {
+      return {
         activeComponent: nextProps.activeComponent || 'loading'
       };
-      switch (state.activeComponent) {
+    },
+
+    /**
+      Generate the content based on the state.
+
+      @method _generateContent
+      @return {Object} The child components for the content.
+    */
+    _generateContent: function() {
```

Comment:
I had to move the content generation out of state so that it can be modified on state updates.